scipy/spatial/distance.py

```diff
@@ -624,6 +631,8 @@ def correlation(u, v, w=None, centered=True):
     """
     u = _validate_vector(u)
     v = _validate_vector(v)
+    _deprecate_complex(u)
+    _deprecate_complex(v)
```

Comment:
For simplicity I would prefer:
```suggestion
    if np.iscomplexobj(u) or np.iscomplexobj(v):
        message = ("Complex `u` and `v` are deprecated and will raise an error in "
                   "SciPy 1.17.0.")
        warnings.warn(message, DeprecationWarning, stacklevel=2)
```
I have also tweaked the warning message

Reply:
Thanks for reviewing! I factored out the `_deprecate_complex` function because I expect we might do the same for the other two dozen of distance functions… but indeed we might as well worry about that later for if we want to do it everywhere we might as well put it into validate_vector.